plugins/woocommerce/src/Admin/API/Reports/Orders/DataStore.php

```diff
@@ -534,6 +542,52 @@ protected function get_coupons_by_order_ids( $order_ids ) {
 		return $coupons;
 	}
 
+	/**
+	 * Get order attributions data from order IDs.
+	 *
+	 * @param array $order_ids Array of order IDs.
+	 * @return array
+	 */
+	protected function get_order_attributions_by_order_ids( $order_ids ) {
+		global $wpdb;
+		$order_meta_table   = OrdersTableDataStore::get_meta_table_name();
+		$included_order_ids = implode( ',', $order_ids );
+
+		if ( OrderUtil::custom_orders_table_usage_is_enabled() ) {
+			/* phpcs:disable WordPress.DB.PreparedSQL.InterpolatedNotPrepared */
+			$order_attributions_meta = $wpdb->get_results(
+				"SELECT order_id, meta_key, meta_value
+					FROM $order_meta_table
+					WHERE order_id IN ({$included_order_ids})
+					AND meta_key IN ('_wc_order_attribution_source_type', '_wc_order_attribution_utm_source')
+					",
+				ARRAY_A
+			);
+			/* phpcs:enable */
+		} else {
+			/* phpcs:disable WordPress.DB.PreparedSQL.InterpolatedNotPrepared */
+			$order_attributions_meta = $wpdb->get_results(
+				"SELECT post_id as order_id, meta_key, meta_value
+					FROM {$wpdb->prefix}postmeta
+					WHERE post_id IN ({$included_order_ids})
+					AND meta_key IN ('_wc_order_attribution_source_type', '_wc_order_attribution_utm_source')
+					",
+				ARRAY_A
+			);
+			/* phpcs:enable */
+		}
+
+		$order_attributions = array();
+		foreach ( $order_attributions_meta as $meta ) {
+			if ( ! isset( $order_attributions[ $meta['order_id'] ] ) ) {
+				$order_attributions[ $meta['order_id'] ] = array();
+			}
+			$order_attributions[ $meta['order_id'] ][ $meta['meta_key'] ] = $meta['meta_value'];
+		}
+
+		return $order_attributions;
+	}
```

Comment:
📜 

I took quite a bit of time to analyse this approach. I was inclined to use `wc_get_orders` instead of direct SQL. The advantages of using `wc_get_orders` are that we don't have to worry about `HPOS` compatibility, we have the `WC_Order` object giving us more information to work with, and our code is a bit cleaner. 

I also reached out to @vedanshujain and asked his opinion. He said both approaches are acceptable.

I did a simple benchmark test using `microtime`, and I can see using direct SQL is slightly faster than `wc_get_orders`. In the context of analytics, I am inclined towards faster results than cleaner code. Let's move forward with this approach and we can tweak it if required at a latest stage.

Reply:
> I was inclined to use `wc_get_orders` instead of direct SQL. [...]

Yes, I thought about using `wc_get_orders` too (reference: https://github.com/woocommerce/woocommerce/wiki/wc_get_orders-and-WC_Order_Query), and I was also thinking about HPOS compatibility.

However, there were two things that I thought about, that I ended up using direct SQL.

One, I'm not sure if `wc_get_orders` might give us too much data ("fat" orders) and cause us issues in production. In our development environment, it's usually pretty simple small order objects; but I'm concerned that in the real world, merchants might be using some plugins or extensions that stores a lot of metadata into an order object. If we use `wc_get_orders`, it would be the first time we retrieve orders object in this Orders Analytics page. Even if the users choose page size 25 (the smallest page size), I'm concerned that it might cause performance issue or "out of memory" issue etc. In other words, I would like to be safe, and use direct SQL to retrieve just the data we need quickly.

Two, I see that we already have existing prior art of using direct SQL in populating other `extended_info` properties (product, customer, and coupon), so I just follow the existing way. 

> I did a simple benchmark test using `microtime` [...]

Just sharing an info that I found in https://www.php.net/manual/en/function.microtime.php: "For performance measurements, using [hrtime()](https://www.php.net/manual/en/function.hrtime.php) is recommended."

But no worry, I guess using `microtime` is fine as a rough gauge. 

> [...] and I can see using direct SQL is slightly faster than `wc_get_orders`.

Related to my comments above about "fat" orders. Using direct SQL is slightly faster in our _development environment_. I'm thinking that in the real world merchant's WooCommerce store production environment, it may be significantly faster compared to using `wc_get_orders`.